### PR TITLE
Add fields - QMAPS-303

### PR DESCRIPTION
**Pull request by jbgriesner**
This PR adds a basic object model structure for blocks returned by the API.
Simple tests have been added (note the json in the fixture corresponds to the bragi answer, but does not corresponds to the es answer that idunn makes).

### CODE REVIEW

File: idunn/blocks/opening_hour.py [hunk 1]
```diff
@@ -86,14 +66,17 @@ def from_es(cls, es_poi, lang):
...
 
         ohr = hoh.render_field(clean_raw, universal=False)
-        description = ohr.full_description()
-        days, is247 = from_desc(description)
 
-        status = ''
+        # plaintext_week = ohr.plaintext_week_description()
+
+        # try:
+        #    days = [{'day_name' d.split(':', 1)[0]: d.split(':', 1)[1].strip()} for d in plaintext_week.splitlines()]
```
Comment (antoine-de) [inline idunn/blocks/opening_hour.py]: exception should be use only sparcingly. check `next`

(and I fhink we should clean this if not used

File: idunn/blocks/opening_hour.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    @classmethod
+    def from_es(cls, es_poi, lang):
+        logger = logging.getLogger('idunn')
+        is247 = False
+
+        raw = es_poi.get('properties', {}).get('opening_hours')
+        if raw is None:
+            return None
+        elif raw is '24/7':
```
Comment (antoine-de) [inline idunn/blocks/opening_hour.py]: you can rewrite this: 
```python
if raw is None:
    return None
is247  = raw == '24/7'
```
(for this use it seems more appropriate to use `==` than `is`)

File: idunn/blocks/opening_hour.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            is247 = True
+
+        try:
+            clean_raw = hoh.OHParser.sanitize(raw)
+            oh = hoh.OHParser(clean_raw)
+        except (SpanOverMidnight, ParseError) as err:
+            logger.error(f"OpeningHour error: {err}")
+            return None
+
+        status = 'close'
```
Comment (antoine-de) [inline idunn/blocks/opening_hour.py]: minor but you can write this `status = 'open' if oh.is_open() else 'close'`

 do we want to use the `open()` function of the library  ? what time is used to compute this ? the server's ? the poi's ?

File: idunn/blocks/opening_hour.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            type='OpeningHourBlock',
+            status=status,
+            next_transition_time=ntt,
+            seconds_before_next_transition=sbnt,
+            is_24_7=is247,
+            raw=clean_raw,
+            days=[] # format definition left for future work
+        )
+
+if __name__ == "__main__":
```
Comment (antoine-de) [inline idunn/blocks/opening_hour.py]: hum why do you need this ?
we should use pytest to run the doc test, not mains

File: tests/test_api.py [hunk 5]
```diff
@@ -33,12 +33,18 @@ def test_basic_query(orsay_museum):
...
+
+    assert resp['id'] == 'osm:way:63178753'
+    assert resp['name'] == "Musee d'Orsay"
+    assert resp['local_name'] == "Musée d'Orsay"
+    assert resp['class_name'] == 'museum'
+    assert resp['subclass_name'] == 'museum'
+    assert resp['address']['label'] == '62B Rue de Lille (Paris)'
+    assert resp['blocks'][0]['btype'] == 'OpeningHourBlock'
+    assert resp['blocks'][0]['is_24_7'] == False
+    assert resp['blocks'][0]['days'][2] == 'Wednesday'
```
Comment (antoine-de) [inline tests/test_api.py]: since we use the `lang`, it would be nice to have a test with a `lang` param

File: idunn/blocks/opening_hour.py (hunk 2 above)
Comment (jbgriesner) [reply idunn/blocks/opening_hour.py]: done

File: idunn/blocks/opening_hour.py (hunk 4 above)
Comment (jbgriesner) [reply idunn/blocks/opening_hour.py]: done

File: tests/test_api.py (hunk 5 above)
Comment (jbgriesner) [reply tests/test_api.py]: done

File: idunn/blocks/opening_hour.py (hunk 3 above)
Comment (jbgriesner) [reply idunn/blocks/opening_hour.py]: They use the present time of the server:
dt = datetime.datetime.now().replace(tzinfo=pytz.UTC)
and check if the corresponding moment belongs to periods of the POI:
https://github.com/rezemika/humanized_opening_hours/blob/63a484ad76a57276823f04240c4ed1af6a516ec6/humanized_opening_hours/temporal_objects.py#L55

I thought it was ok to let investigations of better alternatives as a future work.

File: idunn/api/poi.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                geom = {
+                    'coordinates': [lon, lat],
+                    'type': 'Point'
+                }
+
+        local_name = es_poi.get('properties', {}).get('name')
+        if lang is None:
+            name = local_name
+        else:
+            name = es_poi.get('properties', {}).get(f'name:{lang}')
```
Comment (antoine-de) [inline idunn/api/poi.py]: it does not work if the lang is not in the name.

maybe you should wrap this in a function, this way you can add some doc test on the corner cases (and some comments)

File: tests/test_api.py [hunk 7]
```diff
@@ -33,11 +33,29 @@ def test_basic_query(orsay_museum):
...
+    client = TestClient(app)
+    response = client.get(
+        url=f'http://localhost/v1/pois/{orsay_museum}?lang=it',
+    )
+
+    assert response.status_code == 200
+
+    resp = response.json()
+
+    assert resp['name'] == "Museo d'Orsay"
```
Comment (antoine-de) [inline tests/test_api.py]: I think it's nice to test all the field there too

File: tests/test_api.py [hunk 8]
```diff
@@ -54,6 +72,7 @@ def test_unknow_poi(orsay_museum):
 
 def test_schema():
```
Comment (antoine-de) [inline tests/test_api.py]: you can comment all the test

File: idunn/blocks/opening_hour.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+    @classmethod
+    def from_es(cls, es_poi, lang):
+        logger = logging.getLogger('idunn')
+
+        raw = es_poi.get('properties', {}).get('opening_hours')
+        is247 = False
+        if raw is None:
+            return None
+        elif raw == '24/7':
```
Comment (antoine-de) [inline idunn/blocks/opening_hour.py]: you haven't changed this from my earlier comment

File: idunn/blocks/opening_hour.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+            is247 = True
+
+        return cls(
+            type='OpeningHourBlock',
```
Comment (antoine-de) [inline idunn/blocks/opening_hour.py]: can't we get this from `cls` ?

File: idunn/blocks/block.py [hunk 11]
```diff
@@ -0,0 +1,9 @@
+from apistar import types, validators
+
+class Block(types.Type):
```
Comment (antoine-de) [inline idunn/blocks/block.py]: this class is worthless now no ?

File: idunn/blocks/opening_hour.py (hunk 10 above)
Comment (amatissart) [reply idunn/blocks/opening_hour.py]: Also, the class name should not be used as `type`.
`opening_hours` is expected here in snake case (see the specs).

File: .travis.yml [hunk 12]
```diff
@@ -12,6 +12,7 @@ install:
 
 script:
   - pipenv run pytest tests
+  - pipenv run pytest --doctest-modules
```
Comment (antoine-de) [inline .travis.yml]: I think you can acheive this by adding a configuration file in the project's root. 

this way the doctest will always be run, even in local

File: idunn/api/poi.py [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+
+    'local_name' corresponds to the name in the language of the country where the POI is located.
+
+    >>> get_name({}, 'fr') is None
+    True
+
+    >>> get_name({'name':'spontini', 'name:en':'spontinien', 'name:fr':'spontinifr'}, None)
+    'spontini'
+
+    >>> get_name({'name':'spontini', 'name:en':'spontinien', 'name:fr':'spontinifr'}, 'fr')
```
Comment (antoine-de) [inline idunn/api/poi.py]: a test with a name a an unknown lang would be nice

File: idunn/blocks/opening_hour.py [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+    status = validators.String()
+    next_transition_time = validators.String()
+    seconds_before_next_transition = validators.Integer()
+    is_24_7 = validators.Boolean()
+    raw = validators.String()
+    days = validators.Array()
+
+    @classmethod
+    def from_es(cls, es_poi, lang):
+        logger = logging.getLogger('idunn')
```
Comment (antoine-de) [inline idunn/blocks/opening_hour.py]: the logger is not used

File: .travis.yml [hunk 15]
```diff
@@ -11,7 +11,7 @@ install:
   - pipenv install --dev
 
 script:
-  - pipenv run pytest tests
+  - pipenv run pytest
```
Comment (antoine-de) [inline .travis.yml]: you need to update travis

Comment (antoine-de) [reply .travis.yml]: my bad, I missread
